astarte_device.c

```diff
@@ -687,9 +712,31 @@ static void on_incoming(astarte_device_handle_t device, char *topic, int topic_l
     device->data_event_callback(&event);
 }
 
+static int has_connectivity()
+{
+    esp_http_client_config_t config = {
+        .url = "https://www.google.com",
```

Comment:
we cannot hardcode this address here, I would rather add a config option and default it to connectivitycheck.gstatic.com/generate_204